### PR TITLE
Fix/zig string safety

### PR DESCRIPTION
**Pull request by tarun1sisodia**
# Description

This PR fixes a critical bug in the MetaCall Zig port where string return types were being freed before they could be used by the caller. It introduces a `Value(T)` wrapper pattern that provides manual control over resource deinitialization (RAII-like), ensuring memory safety for multi-language calls.

## Key Changes:
- **root.zig**: Added a `Value(T)` struct with `get()` and `deinit()` methods. Updated `metacall` to return this wrapper for strings while maintaining direct returns for primitive types.
- **Portability**: Replaced hardcoded absolute paths in `build.zig` and `integrated.zig` with repo-relative paths (`../../../build` and `src/tests/*`).
- **Zig Compatibility**: Fixed a reflective type check error (`.Array` vs `.array`) to support Zig 0.16.0-dev.
- **Robustness**: Added integrated stress tests (10,000 calls) and edge case validation for empty strings.

## Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Breaking change (API change for string returns)

# Checklist:
- [X] I have performed a self-review of my own code.
- [X] My changes generate no new warnings.
- [X] I have added tests that prove my fix is effective.
- [X] I have tested the tests pass (`zig build test --summary all`).
- [X] I have run `zig fmt` and followed style guidelines.

### CODE REVIEW

File: source/ports/zig_port/src/tests/integrated.zig [hunk 1]
```diff
@@ -6,10 +6,11 @@ const hi: []const u8 = "hi";
 pub fn main() !void {
     try metacall.init();
 
-    var paths: [1][:0]const u8 = .{"/home/raymond/Projects/metacall-core/source/ports/zig_port/src/tests/test.c"};
+    // Use relative paths for better portability
+    var paths: [1][:0]const u8 = .{"src/tests/test.c"};
     try metacall.load_from_file("c", &paths);
```
[the remark this answers is not in the document]
Comment (trgwii) [reply source/ports/zig_port/src/tests/integrated.zig]: True

**Comment by tarun1sisodia** [comment]
test

above you can see the compiled and tests passed.

and here are the paths i defined 

both images are not using hardcoded path

**Comment by viferga** [comment]
Do you think it's ready to merge then? I would like to add CI even if it's after this PR.

**Comment by tarun1sisodia** [comment]
> Do you think it's ready to merge then? I would like to add CI even if it's after this PR.

Yes ofcourse of you want them merge it or add ci for zig

**Comment by viferga** [comment]
> > Do you think it's ready to merge then? I would like to add CI even if it's after this PR.
> 
> Yes ofcourse of you want them merge it or add ci for zig

I prefer if you add it.

**Comment by viferga** [comment]
@tarun1sisodia can you add zig to the ci? I will leave it as draft meanwhile, check out tools/metacall-environment.sh
